src/databricks/model_training.py

```diff
@@ -274,21 +223,35 @@ def convert_feature_dataframe_to_dict(
 # COMMAND ----------
 
 # DBTITLE 1,Initialize model
-model = SurrogateModel(name="test" if DEBUG else "sf_hvac")
+sm = SurrogateModel(name="test" if DEBUG else "mvp")
 
 # COMMAND ----------
 
-# DBTITLE 1,Fit model
-# Train keras model and log the model with the Feature Engineering in UC.
+# DBTITLE 1,Train model
+# Train keras model and log the model with the Feature Engineering in UC. Note that right we are skipping registering the model in the UC-- this requires storing the signature, which for unclear reasons, is slowing down inference more than 10x.
 
-#Init FeatureEngineering client
+# Init FeatureEngineering client
 fe = FeatureEngineeringClient()
 
 # Set the activation function and numeric data type for the model's layers
-layer_params = {"activation": "leaky_relu", "dtype": np.float32}
+layer_params = {
```

Comment:
Good question. So before we had `layers.Dense(64,  activation = "leaky_relu")(bm)` which first applies the linear mapping and then the non-linear activation. But since we want to apply batchnorm in between the linear map and the activation, we have to instead not apply any non-linear activation at first (i.e., activation='linear' which is the default), then pass it thru a batchnorm layer and then pass it through the leakyrelu layer. There were two additional things that made this confusing that I have now fixed to make this more readable and easier to modify in the future:
1. when you have repeated "blocks" like this (e.g, conv-batchnorm-relu) the standard thing is to define a function for this block rather than just repeat it over and over. I was just being lazy. I have now done this. 
2. I was still using this pattern of manually applying the activation after the dense layer in the combine trunk, even tho I wasn't using batchnorm, so I could have just passed the activation name directly. I removed the activation from the arguments here since this is kind of stupid now, and then updated layers in combine trunk to use the simpler pattern.

Reply:
Ok that makes sense - I'd missed that detail about the ordering of steps here and that the batchnorm function needed to be added in between the linear mapping and non-linear activation. Thanks for adding the function in as well!